src/Mono.Android/Microsoft.Android.Runtime/ManagedValueManager.cs

```diff
@@ -20,52 +22,80 @@ class ManagedValueManager : JniRuntime.JniValueManager
 {
 	const DynamicallyAccessedMemberTypes Constructors = DynamicallyAccessedMemberTypes.PublicConstructors | DynamicallyAccessedMemberTypes.NonPublicConstructors;
 
-	Dictionary<int, List<IJavaPeerable>>?   RegisteredInstances = new Dictionary<int, List<IJavaPeerable>>();
+	readonly Dictionary<int, List<ReferenceTrackingHandle>> RegisteredInstances = new ();
+	readonly ConcurrentQueue<IntPtr> CollectedContexts = new ();
+
+	bool disposed;
+
+	static readonly SemaphoreSlim bridgeProcessingSemaphore = new(1, 1);
```

Comment:
Nit, missing space before parentheses

```suggestion
	static readonly SemaphoreSlim bridgeProcessingSemaphore = new (1, 1);
```